src/test/java/com/puppycrawl/tools/checkstyle/bdd/TestInputConfiguration.java

```diff
@@ -82,11 +82,22 @@ public List<TestInputViolation> getFilteredViolations() {
         return Collections.unmodifiableList(filteredViolations);
     }
 
+    public DefaultConfiguration getTreeWalker() {
+        DefaultConfiguration treeWalker = new DefaultConfiguration(TreeWalker.class.getName());
+        for (ModuleInputConfiguration moduleInputConfiguration : childrenModules) {
+            if (treeWalker.getName().equals(moduleInputConfiguration.getModuleName())) {
+                treeWalker = moduleInputConfiguration.createConfiguration();
+                childrenModules.remove(moduleInputConfiguration);
```

Comment:
Too complicated.

Can you do logic at code around 
```
forEach(moduleConfig -> {
                    if (CHECKER_CHILDREN.contains(moduleConfig.getName())) {
                        root.addChild(
```

Reply:
No problem in that but than I have to copy same code to two functions, will it be ok ?

Reply:
You need to skip Treewalker in foreach

Reply:
What if TreeWalker specified after other check modules than it will be lost if i overwrite.

Reply:
Last Treewalker wins.
You can put Treewalker above Check in header to initialize in required order

Reply:
So, is that compulsory to define the treeWalker at the top ?

Reply:
I am ok with this.
If not defined, we use default.

Reply:
This is last. I got an idea,

While adding child i will check is that treeWalker or not, if it is I will add that at position 0, in this way it always stays at top in list even if defined last in input file, what do you think ?

Reply:
You can sort modules before foreach

Reply:
how about one position variable, which will store the index of treeWalker in list ?

Reply:
Done,

TreeWalker Must Specified At The Top otherwise it takes default treeWalker.